### PR TITLE
Fixes and and use High Sierra's <sys/snapshot.h>.

### PR DESCRIPTION
**Pull request by blair**

### CODE REVIEW

File: snapUtil.c [hunk 1]
```diff
@@ -224,7 +129,7 @@ do_list(const char *vol)
 	}
 
 	char *p = &abuf[0];
-	for (int i; i < count; i++) {
```
Comment (ahl) [inline snapUtil.c:227]: yeesh; embarrassing!

**Review by ahl — APPROVED** [review]
thanks for this!

**Comment by ahl** [comment]
Thanks for the contribution!

**Comment by leftytennis** [comment]
Thank you for this, I had just upgraded to High Sierra and switched to APFS, so I was going to see what it would take to implement snapshot functionality for macOS/APFS!!!!

Going to build from master and try it out, thanks!

**Comment by blair** [comment]
Unfortunately, on High Sierra the system calls are blocked and the program needs an entitlement:

$ sudo ./snapUtil -c test /
Password: 
fs_snapshot_create: Operation not permitted

https://webcache.googleusercontent.com/search?q=cache:LCkhPdsSL2EJ:https://forums.developer.apple.com/thread/89635+&cd=1&hl=en&ct=clnk&gl=us

**Pull request merged**